File: shared/constants/api-routes.ts
/**
 * Centralized API Route Constants
 * ================================
 *
 * Single source of truth for all API routes.
 * Both TypeScript server and Python backend should reference these.
 *
 * Naming Convention:
 * - ROUTE_<DOMAIN>_<ACTION> for specific routes
 * - API_<DOMAIN> for route prefixes
 *
 * Version: 1.0.0 (2025-12-25)
 */

// =============================================================================
// API Versioning
// =============================================================================

export const API_VERSION = 'v1';
export const API_PREFIX = '/api';

// =============================================================================
// Core Domain Routes
// =============================================================================

/**
 * Authentication routes
 */
export const API_AUTH = {
  PREFIX: `${API_PREFIX}/auth`,
  HEALTH: `${API_PREFIX}/auth/health`,
  USER: `${API_PREFIX}/auth/user`,
  LOGIN: `${API_PREFIX}/login`,
  LOGOUT: `${API_PREFIX}/logout`,
} as const;

/**
 * Search routes
 */
export const API_SEARCH = {
  PREFIX: `${API_PREFIX}/search`,
  JOBS: `${API_PREFIX}/search/jobs`,
  START: `${API_PREFIX}/search/start`,
  STOP: `${API_PREFIX}/search/stop`,
  STATUS: `${API_PREFIX}/search/status`,
} as const;

/**
 * Consciousness/QIG routes
 */
export const API_CONSCIOUSNESS = {
  PREFIX: `${API_PREFIX}/consciousness`,
  STATE: `${API_PREFIX}/consciousness/state`,
  METRICS: `${API_PREFIX}/consciousness/metrics`,
  REGIME: `${API_PREFIX}/consciousness/regime`,
} as const;

/**
 * Ocean agent routes
 */
export const API_OCEAN = {
  PREFIX: `${API_PREFIX}/ocean`,
  STATUS: `${API_PREFIX}/ocean/status`,
  MEMORY: `${API_PREFIX}/ocean/memory`,
  NEUROCHEMISTRY: `${API_PREFIX}/ocean/neurochemistry`,
} as const;

/**
 * Olympus (god pantheon) routes
 */
export const API_OLYMPUS = {
  PREFIX: `${API_PREFIX}/olympus`,
  ZEUS: `${API_PREFIX}/olympus/zeus`,
  ATHENA: `${API_PREFIX}/olympus/athena`,
  HERMES: `${API_PREFIX}/olympus/hermes`,
  PANTHEON: `${API_PREFIX}/olympus/pantheon`,
  SHADOW: `${API_PREFIX}/olympus/shadow`,
} as const;

/**
 * Recovery routes
 */
export const API_RECOVERY = {
  PREFIX: `${API_PREFIX}/recovery`,
  UNIFIED: `${API_PREFIX}/unified-recovery`,
  WORKFLOWS: `${API_PREFIX}/recoveries`,
  CANDIDATES: `${API_PREFIX}/recovery/candidates`,
} as const;

/**
 * Balance/blockchain routes
 */
export const API_BALANCE = {
  PREFIX: `${API_PREFIX}/balance`,
  HITS: `${API_PREFIX}/balance-hits`,
  ADDRESSES: `${API_PREFIX}/balance-addresses`,
  MONITOR: `${API_PREFIX}/balance-monitor`,
  QUEUE: `${API_PREFIX}/balance-queue`,
} as const;

/**
 * Vocabulary routes
 */
export const API_VOCABULARY = {
  PREFIX: `${API_PREFIX}/vocabulary`,
  WORDS: `${API_PREFIX}/vocabulary/words`,
  SYNC: `${API_PREFIX}/vocabulary/sync`,
} as const;

// =============================================================================
// Python Backend Routes (QIG-specific)
// =============================================================================

/**
 * QIGGraph search integration routes
 */
export const API_QIGGRAPH = {
  PREFIX: `${API_PREFIX}/search/qiggraph`,
  SCORE: `${API_PREFIX}/search/qiggraph/score`,
  BATCH: `${API_PREFIX}/search/qiggraph/batch`,
  STATE: `${API_PREFIX}/search/qiggraph/state`,
  STATUS: `${API_PREFIX}/search/qiggraph/status`,
} as const;

/**
 * Search tacking routes (kappa-tacking + innate drives)
 */
export const API_TACKING = {
  PREFIX: `${API_PREFIX}/search/tacking`,
  STATUS: `${API_PREFIX}/search/tacking/status`,
  UPDATE: `${API_PREFIX}/search/tacking/update`,
  DRIVES: `${API_PREFIX}/search/tacking/drives`,
} as const;

/**
 * Shadow search bridge routes
 */
export const API_SHADOW_SEARCH = {
  PREFIX: `${API_PREFIX}/shadow/search`,
  STATUS: `${API_PREFIX}/shadow/search/status`,
  BATCH_START: `${API_PREFIX}/shadow/search/batch/start`,
  BATCH_END: `${API_PREFIX}/shadow/search/batch/end`,
  HIGH_PHI: `${API_PREFIX}/shadow/search/high-phi`,
  NEAR_MISS: `${API_PREFIX}/shadow/search/near-miss`,
  MATCH_FOUND: `${API_PREFIX}/shadow/search/match-found`,
} as const;

/**
 * Tokenizer persistence routes
 */
export const API_TOKENIZER = {
  PREFIX: `${API_PREFIX}/tokenizer/pg`,
  STATUS: `${API_PREFIX}/tokenizer/pg/status`,
  SYNC: `${API_PREFIX}/tokenizer/pg/sync`,
  HIGH_PHI: `${API_PREFIX}/tokenizer/pg/high-phi`,
} as const;

// =============================================================================
// Utility Routes
// =============================================================================

export const API_HEALTH = {
  ROOT: '/health',
  API: `${API_PREFIX}/health`,
  FAVICON: '/favicon.ico',
} as const;

export const API_TELEMETRY = {
  PREFIX: `${API_PREFIX}/telemetry`,
  SNAPSHOT: `${API_PREFIX}/telemetry/snapshot`,
  SESSION: `${API_PREFIX}/telemetry/session`,
} as const;

export const API_SELF_HEALING = {
  PREFIX: `${API_PREFIX}/self-healing`,
  STATUS: `${API_PREFIX}/self-healing/status`,
  TRIGGER: `${API_PREFIX}/self-healing/trigger`,
} as const;

// =============================================================================
// Route Collections (for middleware/registration)
// =============================================================================

/**
 * All route prefixes for registration
 */
export const ALL_ROUTE_PREFIXES = [
  API_AUTH.PREFIX,
  API_SEARCH.PREFIX,
  API_CONSCIOUSNESS.PREFIX,
  API_OCEAN.PREFIX,
  API_OLYMPUS.PREFIX,
  API_RECOVERY.PREFIX,
  API_BALANCE.PREFIX,
  API_VOCABULARY.PREFIX,
  API_QIGGRAPH.PREFIX,
  API_TACKING.PREFIX,
  API_SHADOW_SEARCH.PREFIX,
  API_TOKENIZER.PREFIX,
  API_TELEMETRY.PREFIX,
  API_SELF_HEALING.PREFIX,
] as const;

/**
 * Python backend route prefixes (for proxying)
 */
export const PYTHON_BACKEND_ROUTES = [
  API_QIGGRAPH.PREFIX,
  API_TACKING.PREFIX,
  API_SHADOW_SEARCH.PREFIX,
  API_TOKENIZER.PREFIX,
  API_OLYMPUS.PREFIX,
  API_OCEAN.PREFIX,
] as const;

/**
 * Public routes (no auth required)
 */
export const PUBLIC_ROUTES = [
  API_HEALTH.ROOT,
  API_HEALTH.API,
  API_HEALTH.FAVICON,
  API_AUTH.HEALTH,
  API_AUTH.LOGIN,
] as const;

File: qig-backend/api_routes.py
"""
Centralized API Route Constants (Python)
=========================================

Single source of truth for all API routes.
Mirrors the TypeScript constants in shared/constants/api-routes.ts.

IMPORTANT: Keep in sync with TypeScript version!
Run `npm run constants:export` to regenerate if needed.

Version: 1.0.0 (2025-12-25)
"""

from typing import List

# =============================================================================
# API Versioning
# =============================================================================

API_VERSION = "v1"
API_PREFIX = "/api"


# =============================================================================
# Core Domain Routes
# =============================================================================

class ApiAuth:
    """Authentication routes."""
    PREFIX = f"{API_PREFIX}/auth"
    HEALTH = f"{API_PREFIX}/auth/health"
    USER = f"{API_PREFIX}/auth/user"
    LOGIN = f"{API_PREFIX}/login"
    LOGOUT = f"{API_PREFIX}/logout"


class ApiSearch:
    """Search routes."""
    PREFIX = f"{API_PREFIX}/search"
    JOBS = f"{API_PREFIX}/search/jobs"
    START = f"{API_PREFIX}/search/start"
    STOP = f"{API_PREFIX}/search/stop"
    STATUS = f"{API_PREFIX}/search/status"


class ApiConsciousness:
    """Consciousness/QIG routes."""
    PREFIX = f"{API_PREFIX}/consciousness"
    STATE = f"{API_PREFIX}/consciousness/state"
    METRICS = f"{API_PREFIX}/consciousness/metrics"
    REGIME = f"{API_PREFIX}/consciousness/regime"


class ApiOcean:
    """Ocean agent routes."""
    PREFIX = f"{API_PREFIX}/ocean"
    STATUS = f"{API_PREFIX}/ocean/status"
    MEMORY = f"{API_PREFIX}/ocean/memory"
    NEUROCHEMISTRY = f"{API_PREFIX}/ocean/neurochemistry"


class ApiOlympus:
    """Olympus (god pantheon) routes."""
    PREFIX = f"{API_PREFIX}/olympus"
    ZEUS = f"{API_PREFIX}/olympus/zeus"
    ATHENA = f"{API_PREFIX}/olympus/athena"
    HERMES = f"{API_PREFIX}/olympus/hermes"
    PANTHEON = f"{API_PREFIX}/olympus/pantheon"
    SHADOW = f"{API_PREFIX}/olympus/shadow"


class ApiRecovery:
    """Recovery routes."""
    PREFIX = f"{API_PREFIX}/recovery"
    UNIFIED = f"{API_PREFIX}/unified-recovery"
    WORKFLOWS = f"{API_PREFIX}/recoveries"
    CANDIDATES = f"{API_PREFIX}/recovery/candidates"


class ApiBalance:
    """Balance/blockchain routes."""
    PREFIX = f"{API_PREFIX}/balance"
    HITS = f"{API_PREFIX}/balance-hits"
    ADDRESSES = f"{API_PREFIX}/balance-addresses"
    MONITOR = f"{API_PREFIX}/balance-monitor"
    QUEUE = f"{API_PREFIX}/balance-queue"


class ApiVocabulary:
    """Vocabulary routes."""
    PREFIX = f"{API_PREFIX}/vocabulary"
    WORDS = f"{API_PREFIX}/vocabulary/words"
    SYNC = f"{API_PREFIX}/vocabulary/sync"


# =============================================================================
# Python Backend Routes (QIG-specific)
# =============================================================================

class ApiQiggraph:
    """QIGGraph search integration routes."""
    PREFIX = f"{API_PREFIX}/search/qiggraph"
    SCORE = f"{API_PREFIX}/search/qiggraph/score"
    BATCH = f"{API_PREFIX}/search/qiggraph/batch"
    STATE = f"{API_PREFIX}/search/qiggraph/state"
    STATUS = f"{API_PREFIX}/search/qiggraph/status"


class ApiTacking:
    """Search tacking routes (kappa-tacking + innate drives)."""
    PREFIX = f"{API_PREFIX}/search/tacking"
    STATUS = f"{API_PREFIX}/search/tacking/status"
    UPDATE = f"{API_PREFIX}/search/tacking/update"
    DRIVES = f"{API_PREFIX}/search/tacking/drives"


class ApiShadowSearch:
    """Shadow search bridge routes."""
    PREFIX = f"{API_PREFIX}/shadow/search"
    STATUS = f"{API_PREFIX}/shadow/search/status"
    BATCH_START = f"{API_PREFIX}/shadow/search/batch/start"
    BATCH_END = f"{API_PREFIX}/shadow/search/batch/end"
    HIGH_PHI = f"{API_PREFIX}/shadow/search/high-phi"
    NEAR_MISS = f"{API_PREFIX}/shadow/search/near-miss"
    MATCH_FOUND = f"{API_PREFIX}/shadow/search/match-found"


class ApiTokenizer:
    """Tokenizer persistence routes."""
    PREFIX = f"{API_PREFIX}/tokenizer/pg"
    STATUS = f"{API_PREFIX}/tokenizer/pg/status"
    SYNC = f"{API_PREFIX}/tokenizer/pg/sync"
    HIGH_PHI = f"{API_PREFIX}/tokenizer/pg/high-phi"


# =============================================================================
# Utility Routes
# =============================================================================

class ApiHealth:
    """Health check routes."""
    ROOT = "/health"
    API = f"{API_PREFIX}/health"
    FAVICON = "/favicon.ico"


class ApiTelemetry:
    """Telemetry routes."""
    PREFIX = f"{API_PREFIX}/telemetry"
    SNAPSHOT = f"{API_PREFIX}/telemetry/snapshot"
    SESSION = f"{API_PREFIX}/telemetry/session"


class ApiSelfHealing:
    """Self-healing routes."""
    PREFIX = f"{API_PREFIX}/self-healing"
    STATUS = f"{API_PREFIX}/self-healing/status"
    TRIGGER = f"{API_PREFIX}/self-healing/trigger"


# =============================================================================
# Route Collections
# =============================================================================

ALL_ROUTE_PREFIXES: List[str] = [
    ApiAuth.PREFIX,
    ApiSearch.PREFIX,
    ApiConsciousness.PREFIX,
    ApiOcean.PREFIX,
    ApiOlympus.PREFIX,
    ApiRecovery.PREFIX,
    ApiBalance.PREFIX,
    ApiVocabulary.PREFIX,
    ApiQiggraph.PREFIX,
    ApiTacking.PREFIX,
    ApiShadowSearch.PREFIX,
    ApiTokenizer.PREFIX,
    ApiTelemetry.PREFIX,
    ApiSelfHealing.PREFIX,
]

PYTHON_BACKEND_ROUTES: List[str] = [
    ApiQiggraph.PREFIX,
    ApiTacking.PREFIX,
    ApiShadowSearch.PREFIX,
    ApiTokenizer.PREFIX,
    ApiOlympus.PREFIX,
    ApiOcean.PREFIX,
]

PUBLIC_ROUTES: List[str] = [
    ApiHealth.ROOT,
    ApiHealth.API,
    ApiHealth.FAVICON,
    ApiAuth.HEALTH,
    ApiAuth.LOGIN,
]


# =============================================================================
# Convenience Exports
# =============================================================================

# For backwards compatibility with existing code
API_AUTH = ApiAuth
API_SEARCH = ApiSearch
API_CONSCIOUSNESS = ApiConsciousness
API_OCEAN = ApiOcean
API_OLYMPUS = ApiOlympus
API_RECOVERY = ApiRecovery
API_BALANCE = ApiBalance
API_VOCABULARY = ApiVocabulary
API_QIGGRAPH = ApiQiggraph
API_TACKING = ApiTacking
API_SHADOW_SEARCH = ApiShadowSearch
API_TOKENIZER = ApiTokenizer
API_HEALTH = ApiHealth
API_TELEMETRY = ApiTelemetry
API_SELF_HEALING = ApiSelfHealing
